README.md

```diff
@@ -0,0 +1,42 @@
+stack-to-nix
+===========
+
+This is a handy nix function for building stack projects with nix, fully inside the nix sandbox.
+It parses the stack.yaml file and translates the dependencies into a series of fetchUrl, fetchGit calls.
+
+## Example
+
+```nix
+{ pkgs }:
+
+let stackToNix = pkgs.callPackage (fetchTarball https://github.com/serokell/stack-to-nix/archive/master.tar.gz) { };
+in
+stackToNix {
+  # root: the path with stack.yaml. you may want to filter this. for example using nix-gitignore.
+  root = ./.;
+  # shell: get the .env instead of the nix-build derivation. recommended that you do this with shell.nix/default.nix.
+  # see https://github.com/DisciplinaOU/disciplina/blob/master/shell.nix
+  shell = pkgs.lib.inNixShell;
+  # you shouldn't need overrides, but you can ;)
+  overrides = final: previous: with pkgs.haskell.lib; {
+    qtah = overrideCabal previous.qtah (super: {
+      libraryToolDepends = with pkgs.qt5; [ qtbase qttools ];
+    });
+  };
+}
+```
+
+## Problems
+
+This depends on [NixOS/nix#2409](https://github.com/NixOS/nix/pull/2409). It's in our patch set. `nix-env -f https://github.com/serokell/serokell-closure/archive/master.tar.gz -iA nix`
```

Comment:
This is not completely true: it only needs that patch if any of the Git dependencies use revs that are not on HEAD.

Reply:
Also, this would apply to the same extent to all other foo-to-nix tools. This is as much of a problem for mix-to-nix, for example.

Reply:
It will specify `ref="*";` even if they are.